src/cljs/ataru/virkailija/application/handlers.cljs

```diff
@@ -90,14 +92,19 @@
                  (assoc-in [:application :applications] applications)
                  (assoc-in [:application :fetching-applications] false)
                  (assoc-in [:application :review-state-counts] (review-state-counts applications))
-                 (assoc-in [:application :sort] application-sorting/initial-sort))]
+                 (assoc-in [:application :sort] application-sorting/initial-sort))
+          application-key (if (= 1 (count applications))
+                            (-> applications first :key)
+                            (:application-key (cljs-util/extract-query-params)))]
       {:db       db
-       :dispatch (if (= (count applications) 1)
-                   [:application/select-application (-> applications first :key)]
+       :dispatch (if application-key
+                   [:application/select-application application-key]
                    [:application/close-application])})))
 
 (defn fetch-applications-fx [db path]
-  {:db   (assoc-in db [:application :fetching-applications] true)
+  {:db   (-> db
+             (assoc-in [:application :fetching-applications] true)
+             (assoc-in [:application :filter] (cljs-util/get-unselected-review-states (list (:unselected-states (cljs-util/extract-query-params))))))
```

Comment:
Looks like we need to split the query-param value here, like `(assoc-in [:application :filter] (cljs-util/get-unselected-review-states (-> (cljs-util/extract-query-params) :unselected-states (clojure.string/split ","))))` for the filters to work.

Also passing `unselected-states` as an argument named `selected-states` here seems weird?

Reply:
Fixed those.